apps/dashboard/src/pages/vercel-integration-page.tsx

```diff
@@ -0,0 +1,115 @@
+import { useMemo } from 'react';
+import { useOrganization, useOrganizationList } from '@clerk/clerk-react';
+
+import { DashboardLayout } from '@/components/dashboard-layout';
+import { Card, CardContent, CardHeader } from '@/components/primitives/card';
+import { Skeleton } from '@/components/primitives/skeleton';
+import { useVercelIntegrationDetails } from '@/hooks/use-vercel-integration-details';
+import { useVercelIntegration } from '@/hooks/use-vercel-integration-setup';
+import { useVercelParams } from '@/hooks/use-vercel-params';
+import { VercelIntegrationForm } from '@/components/vercel-integration-form';
+import { useVercelProjects } from '@/hooks/use-vercel-projects';
+import { useEnvironment } from '@/context/environment/hooks';
+
+export const VercelIntegrationPage = () => {
+  const { currentEnvironment } = useEnvironment();
+  const { organization } = useOrganization();
+  const { userMemberships } = useOrganizationList({
+    userMemberships: { infinite: true },
+  });
+  const { configurationId, next, isEditMode } = useVercelParams();
+  const { isPending: isVercelIntegrationPending, data } = useVercelIntegration();
```

Comment:
start the Vercel integration on mount, that will create a config for the current org